demo_nodes_cpp/src/logging/use_logger_service.cpp

```diff
@@ -0,0 +1,220 @@
+        RCLCPP_WARN(this->get_logger(), "%s with WARN logger level.", msg->data.c_str());
+        RCLCPP_ERROR(this->get_logger(), "%s with ERROR logger level.", msg->data.c_str());
+      };
+
+    sub_ = this->create_subscription<std_msgs::msg::String>("output", 10, callback);
+  }
+
+private:
+  rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub_;
+};
+
+class TestNode : public rclcpp::Node
+{
+public:
+  explicit TestNode(const std::string & remote_node_name)
+  : Node("TestNode"),
+    remote_node_name_(remote_node_name)
+  {
+    pub_ = this->create_publisher<std_msgs::msg::String>("output", 10);
+    logger_set_client_ = this->create_client<rcl_interfaces::srv::SetLoggerLevels>(
+      remote_node_name + "/set_logger_levels");
+    logger_get_client_ = this->create_client<rcl_interfaces::srv::GetLoggerLevels>(
+      remote_node_name + "/get_logger_levels");
+  }
+
+  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr get_pub()
+  {
+    return pub_;
+  }
+
+  bool set_logger_level_on_remote_node(
+    rclcpp::Logger::Level logger_level)
+  {
+    if (!logger_set_client_->wait_for_service(2s)) {
+      return false;
+    }
+
+    auto request = std::make_shared<rcl_interfaces::srv::SetLoggerLevels::Request>();
+    auto set_logger_level = rcl_interfaces::msg::LoggerLevel();
+    set_logger_level.name = remote_node_name_;
+    set_logger_level.level = static_cast<uint32_t>(logger_level);
+    request->levels.emplace_back(set_logger_level);
+
+    auto result = logger_set_client_->async_send_request(request);
+
+    if (rclcpp::spin_until_future_complete(this->shared_from_this(), result) !=
+      rclcpp::FutureReturnCode::SUCCESS)
+    {
+      return false;
+    }
+
+    auto ret_result = result.get();
+    if (!ret_result->results[0].successful) {
+      RCLCPP_ERROR(
+        this->get_logger(), "Failed to change logger level: %s",
+        ret_result->results[0].reason.c_str());
+      return false;
+    }
+    return true;
+  }
+
+  bool get_logger_level_on_remote_node(uint32_t & level)
+  {
+    if (!logger_get_client_->wait_for_service(2s)) {
+      return false;
+    }
+
+    auto request = std::make_shared<rcl_interfaces::srv::GetLoggerLevels::Request>();
+    request->names.emplace_back(remote_node_name_);
+    auto result = logger_get_client_->async_send_request(request);
+    if (rclcpp::spin_until_future_complete(shared_from_this(), result) !=
+      rclcpp::FutureReturnCode::SUCCESS)
+    {
+      return false;
+    }
+
+    auto ret_result = result.get();
+    level = ret_result->levels[0].level;
+    return true;
+  }
+
+private:
+  const std::string remote_node_name_;
+  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub_;
+  rclcpp::Client<rcl_interfaces::srv::SetLoggerLevels>::SharedPtr logger_set_client_;
+  rclcpp::Client<rcl_interfaces::srv::GetLoggerLevels>::SharedPtr logger_get_client_;
+};
+
+
+int main(int argc, char ** argv)
+{
+  setvbuf(stdout, NULL, _IONBF, BUFSIZ);
```

Comment:
I don't believe that we need this anymore (it is an artifact of long ago before we printed all log messages to stderr).

Reply:
Removed it